src/planners/rbt/RGBTConnect.cpp

```diff
@@ -97,20 +97,20 @@ std::tuple<base::State::Status, std::shared_ptr<std::vector<std::shared_ptr<base
 {
     float d_c = ss->computeDistance(q);
 	std::shared_ptr<base::State> q_new = q;
-	std::shared_ptr<std::vector<std::shared_ptr<base::State>>> q_new_list = std::make_shared<std::vector<std::shared_ptr<base::State>>>();
+	std::vector<std::shared_ptr<base::State>> q_new_list;
```

Comment:
Initialize to fixed size, then change emplace_back in for loop to .at()

Reply:
Thanks for the comment. I have tried, but the problem is when `q_new_list` does not contain `RGBTConnectConfig::NUM_LAYERS` elements, because of premature termination in the loop. Later, I need to use all states from `q_new_list`.